src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MirrorEndpointsSpec.groovy

```diff
@@ -79,57 +91,55 @@ class MirrorEndpointsSpec extends HealthCheckSpecification {
     def "Able to CRUD a mirror endpoint on the src switch, mirror to the same switch diff port [#swPair.src.hwSwString, #mirrorDirection]#trafficDisclaimer"() {
         given: "A flow"
         assumeTrue(swPair as boolean, "Unable to find a switch pair")
-        def flow = flowHelperV2.randomFlow(swPair).tap { maximumBandwidth = 100000 }
+        def flowEntity = flowFactory.getBuilder(swPair).withBandwidth(100000)
         if (profile == 'virtual') {
             // ovs switch doesn't support mirroring for the vxlan flows
-            flow.encapsulationType = FlowEncapsulationType.TRANSIT_VLAN
+            flowEntity.withEncapsulationType(FlowEncapsulationType.TRANSIT_VLAN)
         }
-        flowHelperV2.addFlow(flow)
+        def flow = flowEntity.build().create()
 
         when: "Create a mirror point on src switch, pointing to a different port, random vlan"
+        def flowEp = flow.deepCopy().source
         def mirrorTg = swPair.src.traffGens ?[1]
-        def mirrorPort = mirrorTg?.switchPort ?: (topology.getAllowedPortsForSwitch(swPair.src) - flow.source.portNumber)[0]
-        def mirrorEndpoint = FlowMirrorPointPayload.builder()
-                .mirrorPointId(flowHelperV2.generateFlowId())
-                .mirrorPointDirection(mirrorDirection.toString().toLowerCase())
-                .mirrorPointSwitchId(swPair.src.dpId)
-                .sinkEndpoint(FlowEndpointV2.builder().switchId(swPair.src.dpId).portNumber(mirrorPort)
-                        .vlanId(flowHelperV2.randomVlan())
-                        .build())
-                .build()
-        northboundV2.createMirrorPoint(flow.flowId, mirrorEndpoint)
+        def mirrorPort = mirrorTg?.switchPort ?: (topology.getAllowedPortsForSwitch(swPair.src) - flowEp.portNumber)[0]
+        def mirrorPointEp = flowEp.tap {
+            it.portNumber = mirrorPort
+            it.vlanId = randomVlan()
+        }
+        def mirrorPointPayload = flow.createMirrorPointPayload(mirrorPointEp, mirrorDirection)
+        flow.createMirrorPointWithPayload(mirrorPointPayload)
 
         then: "Mirror status changes to Active"
         Wrappers.wait(WAIT_OFFSET) {
-            assert northboundV2.getFlow(flow.flowId).mirrorPointStatuses[0].status ==
+            assert flow.retrieveDetails().mirrorPointStatuses[0].status ==
                     FlowPathStatus.ACTIVE.toString().toLowerCase()
         }
 
         and: "Flow history reports a successful mirror creation"
         Wrappers.wait(WAIT_OFFSET) {
```

Comment:
Yep, this is more compact, but here we check the status of the created mirror point, the purpose of the test is CRUD of the mirror point, so I think it's better to get the Mirror point and check its status directly rather than through the history (the history check is below in the test).

Reply:
Done